Integrations/Jira/Jira.py

```diff
@@ -0,0 +1,651 @@
+
+    md_and_context = generate_md_context_get_issue(j_res)
+    human_readable = tableToMarkdown(demisto.command(), md_and_context['md'], argToList(headers))
+    if is_update:
+        human_readable += 'Issue #{} was updated successfully'.format(issue_id)
+
+    contents = j_res
+    outputs = {'Ticket(val.Id == obj.Id)': md_and_context['context']}
+    return_outputs(readable_output=human_readable, outputs=outputs, raw_response=contents)
+
+
+@logger
+def issue_query_command(query, start_at='', max_results=None, headers=''):
+    j_res = run_query(query, start_at, max_results)
+    issues = demisto.get(j_res, 'issues')
+    md_and_context = generate_md_context_get_issue(issues)
+    human_readable = tableToMarkdown(demisto.command(), md_and_context['md'], argToList(headers))
+    contents = j_res
+    outputs = {'Ticket(val.Id == obj.Id)': md_and_context['context']}
+    return_outputs(readable_output=human_readable, outputs=outputs, raw_response=contents)
+
+
+@logger
+def create_issue_command():
+    url = 'rest/api/latest/issue'
+    issue = get_issue_fields(**demisto.args())
+    result = jira_req('POST', url, json.dumps(issue))
+    j_res = result.json()
+
+    md_and_context = generate_md_context_create_issue(j_res, project_key=demisto.args().get('projectKey'),
+                                                      project_name=demisto.args().get('issueTypeName'))
+    human_readable = tableToMarkdown(demisto.command(), md_and_context['md'], "")
+    contents = j_res
+    outputs = md_and_context['context']
+    return_outputs(readable_output=human_readable, outputs=outputs, raw_response=contents)
+
+
+@logger
+def edit_issue_command(issue_id, headers=None, status=None, **kwargs):
+    url = 'rest/api/latest/issue/{issue_id}/'.format(issue_id=issue_id)
+    issue = get_issue_fields(**demisto.args())
+    jira_req('PUT', url, json.dumps(issue))
+    if status:
+        edit_status(issue_id, status)
+    return get_issue(issue_id, headers, is_update=True)
+
+
+@logger
+def edit_status(issue_id, status):
+    # check for all authorized transitions available for this user
+    # if the requested transition is available, execute it.
+    url = 'rest/api/2/issue/{issue_id}/transitions'.format(issue_id=issue_id)
+    result = jira_req('GET', url)
+    j_res = result.json()
+    transitions = [transition.get('name') for transition in j_res.get('transitions')]
+    for i, transition in enumerate(transitions):
+        if transition.lower() == status.lower():
+            url = 'rest/api/latest/issue/{issue_id}/transitions?expand=transitions.fields'.format(issue_id=issue_id)
+            json_body = {"transition": {"id": str(j_res.get('transitions')[i].get('id'))}}
+            return jira_req('POST', url, json.dumps(json_body))
+
+    return_error('Status "{}" not found. \n'
+                 'Valid transitions are: {} \n'.format(status, transitions))
+
+
+@logger
+def get_comments_command(issue_id):
```

Comment:
this is a general change request to all of the command functions.
the `foo_command()` function should not have any arguments.
It should extract all of the relevant argents in the beginning, call a business function that will return a raw response,
and then the `foo_command()` take the response and create context outputs and human readable.(possibly with another helper function)

please see the code convention documentation for command functions and fix accordingly:
https://github.com/demisto/content/blob/master/docs/code_conventions/README.MD#command-functions